lib/OpenQA/Setup.pm

```diff
@@ -52,7 +52,8 @@ sub read_config {
             job_investigate_git_timeout => 20,
         },
         rate_limits => {
-            search => 5,
+            results_max => 1000,
```

Comment:
With your changes rendering doesn't seem to be the bottle neck anymore and the spinner works also quite nicely. It still makes generally sense to limit it. However, here it seems to handle 37873 matches quite well. So I guess we can put 40000 or even 50000 as upper limit.

I also wouldn't call it just `results_max` because "results" is very ambiguous in the context of openQA. What about `search_result_limit`? And having this within the `rate_limits` section also seems odd because it isn't really about the rate. Alternatively you could also just accept an URL query parameter `limit` like it is done by other routes as well.

Reply:
I was seeing this as a safeguard first of all, to avoid bogging down the server, rather than a limit users would like - so if we wanted a `limit` parameter it could only be equal to or lower than the config option. This is also why I thought the same category might fit. It's a hard limit on possible search results.

Reply:
Ok, having this as a safeguard make sense. The category is still not fitting in my opinion. This is not the rate limit for the `results_max` route. In fact, it has nothing to do with time/durations and is therefore also not about a rate. And I'd still increase the upper limit.